### PR TITLE
[issue/87] Enable board announcements for new users. Fixes #87.

### LINKED ISSUE #87: New Users Don't See Current Announcement
When a new user is created, their board_announcement_status is 0 by default. So BA thinks they dismissed the announcement, so they don't see it anymore after they registered.

We need to set the users board_announcements_status to 1 when a user is created if board announcements config is enabled.

@rxu I think you've done something like this before, with an event during user creation.

https://www.phpbb.com/customise/db/extension/boardannouncements/support/topic/146966

### PR DESCRIPTION
**Pull request by rxu**

### CODE REVIEW

File: migrations/v10x/m5_enable_announcements_for_new_users.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+*
+* Board Announcements extension for the phpBB Forum Software package.
+*
+* @copyright (c) 2014 phpBB Limited <https://www.phpbb.com>
```
Comment (VSEphpbb) [inline migrations/v10x/m5_enable_announcements_for_new_users.php]: year should be 2015

**Comment by VSEphpbb** [comment]
Are you up to adding a functional test for this? One that creates a new user, logs in as the new user and then checks if the announcement is present.

File: migrations/v10x/m5_enable_announcements_for_new_users.php (hunk 1 above)
Comment (rxu) [inline migrations/v10x/m5_enable_announcements_for_new_users.php]: Heck, I thought I was perfect just like you :D

**Comment by rxu** [comment]
Yeah, I was thinking about adding tests, will look into it when get some free time (in a day or so).

File: migrations/v10x/m5_enable_announcements_for_new_users.php (hunk 1 above)
Comment (VSEphpbb) [inline migrations/v10x/m5_enable_announcements_for_new_users.php]: I thought you were testing me

**Comment by VSEphpbb** [comment]
It's super simple:

https://github.com/phpbb-extensions/boardrules/blob/master/tests/functional/viewonline_test.php#L39-L41
and
https://github.com/phpbb-extensions/boardannouncements/blob/master/tests/functional/announcement_test.php#L74-L75

You can just add the test after `test_view_as_user()` in the functional announcement_test file. Just remember to also log_out first (like in `test_view_as_guest()` before logging in as the new user)

**Comment by rxu** [comment]
Updated #88.

**Pull request merged**